### PR TITLE
[CN-359] Set HotBackup status to Pending after creation and before the start

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: controllers/hazelcast/hot_backup_controller.go [hunk 1]
```diff
@@ -268,8 +276,18 @@ func (r *HotBackupReconciler) executeFinalizer(ctx context.Context, hb *hazelcas
...
+func (r *HotBackupReconciler) triggerHotBackup(ctx context.Context, req reconcile.Request, rest *RestClient, logger logr.Logger) error {
+	hb := &hazelcastv1alpha1.HotBackup{}
+	err := r.Get(ctx, req.NamespacedName, hb)
+	if err != nil {
+		return err
+	}
+	if !hb.Status.State.IsRunning() {
+		hb.Status.State = hazelcastv1alpha1.HotBackupPending
+		_ = r.Status().Update(ctx, hb)
+	}
```
Comment (mtyazici) [inline controllers/hazelcast/hot_backup_controller.go:288]: Why are we not using the existing hot backup object from the reconciler function?

Comment (SeriyBg) [reply controllers/hazelcast/hot_backup_controller.go:288]: The reason is that the `HotBackup` object might be updated by the status update function, thus we need to fetch the last version of it before the update

Comment (mtyazici) [reply controllers/hazelcast/hot_backup_controller.go:288]: I see, this only happens when HotBackup is scheduled with a cron job, right? Also, what happens when user removes the cron job from the CR? From what I get, the cron job will continue and HotBackup also will get triggered once for the change.

**Review by dzeromski-hazelcast — APPROVED** [review]

File: controllers/hazelcast/hot_backup_controller_test.go [hunk 2]
```diff
@@ -136,13 +137,80 @@ func TestHotBackupReconciler_shouldSetStatusToFailedWhenHbCallFails(t *testing.T
...
+
+	Eventually(func() hazelcastv1alpha1.HotBackupState {
+		_ = r.Client.Get(context.TODO(), n, hb)
+		return hb.Status.State
+	}, 2*time.Second, 100*time.Millisecond).Should(Equal(hazelcastv1alpha1.HotBackupPending))
+
+	reconcileWg.Add(1)
+	go func() {
+		defer reconcileWg.Done()
+		_, _ = r.Reconcile(context.TODO(), reconcile.Request{NamespacedName: n})
```
Comment (dzeromski-hazelcast) [inline controllers/hazelcast/hot_backup_controller_test.go:206]: nit: skip the assignment?

Comment (SeriyBg) [reply controllers/hazelcast/hot_backup_controller_test.go:206]: We can skip it, but Goland told me not to do so 😃 
image

**Review by mtyazici — APPROVED** [review]

**Pull request merged**